src/server/service/parking-lot.ts

```diff
@@ -0,0 +1,56 @@
+import ParkingLotModel from '../models/parking-lot';
+import { UnexistentParkingLotID } from '../error/parking-lot.service';
+
+import type { ParkingLot } from '../models/parking-lot';
+
+export type CreateParkingLotDTO = {
+  name: string;
+};
+
+export type UpdateParkingLotDTO = {
+  id: string;
+  name?: string;
+};
+
+export interface IParkingLotService {
+  create(dto: CreateParkingLotDTO): Promise<ParkingLot>;
+  update(dto: UpdateParkingLotDTO): Promise<ParkingLot>;
+  remove(id: string): Promise<void>;
+  findById(id: string): Promise<ParkingLot | null>;
+}
+
+export default class ParkingLotService implements IParkingLotService {
+  async create(dto: CreateParkingLotDTO): Promise<ParkingLot> {
+    const parkingLot = new ParkingLotModel(dto);
+
+    await parkingLot.save();
+
+    return parkingLot;
+  }
+
+  async update(dto: UpdateParkingLotDTO): Promise<ParkingLot> {
+    const parkingLot = await ParkingLotModel.findById(dto.id);
+
+    await parkingLot.update(dto);
+
+    return parkingLot;
+  }
+
+  async remove(id: string): Promise<void> {
+    const parkingLot = await ParkingLotModel.findById(id);
+
+    await parkingLot.remove();
```

Comment:
Aren't we doing soft deletes? Perhaps this could be just an update on a `deleted_at` property for the ParkingLot document.

Reply:
Also, what do you think about having a `created_at` and `updated_at` properties for this model and the User model as well?

Reply:
Is a good idea to have a deleted_at field?